### PR TITLE
DE-6489: Clean up props

### PR DESCRIPTION
**Pull request by fingerartur**
**_This PR is all about simplicity._**

Here I mostly focused on simplifying the code and some of the public props.

In my opinion `Player` and `clpp.Player`, etc. should be in React context since almost all child components rely on them. This lead to a nice simplification of both the code and the API (props).

I reduced direct manipulation of the DOM in a few places (direct manipulation is always dangerous if used in tandem with React).

Lastly I upgraded test tooling, which will speed up future writing of tests.

_See the commit message for a full list of goodies._

### CODE REVIEW

**Review by thasso — APPROVED** [review]
Looks good generally and I think it does make usage simpler.

If I see it correctly, we also have a way to use components outside of the PlayerSurface, which is what we do in `CustomControlsPage`. Maybe we can add that as a basic example to the Readme?

**Comment by fingerartur** [comment]
@thasso Correct, we do (but I think you meant `app/src/ComponentsOverviewPage.tsx`). I'm gonna focus on documentation and code examples in https://castlabs.atlassian.net/browse/DE-6500, so I'd leave it for then.

File: src/components/Duration.tsx [hunk 1]
```diff
@@ -2,20 +2,25 @@ import React, { useState } from 'react'
...
 
 import { Label } from './Label'
 
-export type DurationProps = BasePlayerComponentProps
+export interface DurationProps extends BaseComponentProps {
+  children?: React.ReactNode
+}
 
+/**
+ * Duration.
```
Comment (doubeka) [inline src/components/Duration.tsx:17]: Do we want to add here more details? Or I am maybe missing point of this description?
There are few other similar descriptions. Maybe we can followup on these in documentation task.

**Review by doubeka — APPROVED** [review]
I agree it is great usage of Context here for simplification. 
Also nice cleanup on test ids and start on separation of snapshots.

File: src/components/Duration.tsx (hunk 1 above)
Comment (fingerartur) [reply src/components/Duration.tsx:17]: Totally agree, my problem is that I actually don't know yet what every component does :D, once I find out, I'll improve the descriptions.

**Pull request merged**